server.go

```diff
@@ -220,7 +225,7 @@ func (sc *serverConn) handleStreams() {
 				continue
 			}
 
-			if len(strms) >= int(sc.st.maxStreams) {
+			if len(strms)+len(closedStrms) >= int(sc.st.maxStreams) {
```

Comment:
len(closedStrms)? Why do we keep the closedStrms? That can grow infinitely

Reply:
We need to be able to identify closed stream to send a closed stream error when frame arrive on those streams.

Reply:
Ok, I am fine with that, but the maxStreams is the maximum current open streams, right? So, if we do `current_streams + closed_streams >= max_streams` it's going to fail as soon as the client opened more than 128 streams, although the client might have only one stream open at that time.

Reply:
Yes, You're right, the fix was not the right one but a consequence of another bug around `SETTINGS_INITIAL_WINDOW_SIZE`.

So I remove it and I will do another PR with the fix on `SETTINGS_INITIAL_WINDOW_SIZE` to fix this spec.